Dockerfile

```diff
@@ -0,0 +1,17 @@
+FROM elixir:1.7.3-slim
+
+RUN adduser --disabled-password --gecos '' test
+USER test
+
+RUN mix local.hex --force
+RUN mix local.rebar --force
+
+COPY --chown=test:test . /code
+WORKDIR /code
```

Comment:
You can significantly reduce build times on Jenkins by mounting the code as a volume in the `docker-compose.yml` instead of copying it over every time.

Reply:
If I use a volume for /code I get this error:
```
  1) test it requeues messages when processing function does not return :ok (ExKonsument.ConsumerTest)
     test/exkonsument/consumer_test.exs:40
     No message matching {:EXIT, ^pid, {%ExKonsument.HandlingError{return: :not_ok}, _}} after 100ms.
     The following variables were pinned:
       pid = #PID<0.464.0>
     Process mailbox:
       {:open_channel, #PID<0.463.0>}
       :bind_queue
     code: assert_receive {
     stacktrace:
       test/exkonsument/consumer_test.exs:61: (test)
```

Reply:
I think this is just because the code runs a little slower in a volume. You can probably get around that by passing a greater timeout to `assert_receive`, like one second. The build will still be significantly faster, because without the volume you have to recompile the whole code base on every Jenkins build.